### PR TITLE
Library migration: => `file` and `mockito`.

### PR DESCRIPTION
**Pull request by pq**
WIP (not ready for landing).

### CODE REVIEW

**Comment by pq** [comment]
@srawlins: would you mind taking a gander at my migration to mockito?

**Comment by srawlins** [comment]
Mocks LGTM 😄

**Comment by nilsdoehring** [comment]
@pq: Thanks for this, but unsure how to proceed. pub get fails (see travis build):

> Package mockito has no versions that match >=2.2.3 <3.0.0 derived from:
> \- dart_coveralls depends on version ^2.2.3

**Comment by srawlins** [comment]
Is it the Dart SDK constraint? 2.2.3 requires Dart `>=2.0.0-dev.16.0 <2.0.0`.

**Comment by pq** [comment]
@srawlins : I notice that mockito doesn't even try to build on stable...  Will that change when stable bumps to 2.0.0?

**Comment by srawlins** [comment]
@pq what do you mean, "doesn't even try"?

**Comment by pq** [comment]
@srawlins: sorry...  I mean in the mockito [travis config](https://github.com/dart-lang/mockito/blob/master/.travis.yml#L22) you only build against dev whereas [here](https://github.com/block-forest/dart-coveralls/blob/master/.travis.yml#L3-L5) dart-coveralls does stable and dev:

```
dart:
  - stable
  - dev
```

**Comment by pq** [comment]
And so, unsurprisingly, we see failures on stable but not dev for this PR:

image

**Comment by nilsdoehring** [comment]
From my POV, I think it's fine to stop maintaining this package for Dart 1.x, making the current version the last one supporting it. 

With this PR we would then have to:

- [] add a >2.0.0-dev SDK constraint in pubspec*
- [] bump the version in pubspec
- [] add a disclaimer to README.md explaining the version situation
- [] add the new builder dependencies to pubspec
- [] add a build.yaml (if necessary)
- [] add .dart-tool/ to .git-ignore
- [] change the travis config to not even try to build with SDK 1.x

Thoughts?

\* Do we know how _pub get_ behaves when run in SDK v1.x on a package with a dependency to _dart_coveralls: any_ (which would state an SDK 2.x constraint)? Will it break or will it find the most recent package that has a matching SDK constraint and use that?

**Comment by pq** [comment]
> Do we know how pub get behaves when run in SDK v1.x on a package with a dependency to dart_coveralls: any (which would state an SDK 2.x constraint)? Will it break or will it find the most recent package that has a matching SDK constraint and use that?

I'm not really sure.  I think it should find the most recent matching package. @kevmoo ?

As for your punchlist of proposed changes, they look good to me.  I'm missing some context on these:

> [] add the new builder dependencies to pubspec
> [] add a build.yaml (if necessary)

Otherwise, would you like me to start adding changes?

Thanks for following up!

**Comment by kevmoo** [comment]
> Will it break or will it find the most recent package that has a matching SDK constraint and use that

Yes!

**Comment by nilsdoehring** [comment]
@pq Now that you mention it, I'm missing context for those two items, too – they're only required for webdev projects ... my bad.

I'd be delighted if you could add these changes to your PR. Thanks much!

> Will it break **or** will it find the most recent package that has a matching SDK constraint and use that
> 
> Yes!

@kevmoo cool, it will break AND work at the same time ;-) 
(I take it you were referring to the latter and it will work)

**Comment by pq** [comment]
@nilsdoehring: apologies for the slow follow-up!

* [ ] add a >2.0.0-dev SDK constraint in pubspec*
* [ ] bump the version in pubspec
* [ ] add a disclaimer to README.md explaining the version situation
* [ ] add .dart-tool/ to .git-ignore
* [ ] change the travis config to not even try to build with SDK 1.x

We should be good to go.

PTAL.

File: test/coveralls_test.dart [hunk 1]
```diff
@@ -55,23 +53,24 @@ void main() {
...
-          .thenReturn(new Future.value(processResult));
-      var remotes =
-          GitRemote.getGitRemotes(mockDir, processSystem: processSystem);
-      expect(remotes.length, equals(1));
-      expect(remotes.single.name, equals("origin"));
-      expect(remotes.single.address,
-          equals("git@github.com:Adracus/dart-coveralls.git"));
+      when(processResult.exitCode).thenReturn(0);
+
+      //TODO(pq): fix stubbing here.
```
Comment (nilsdoehring) [inline test/coveralls_test.dart]: @pq want to keep this here for now? Fine for me.

**Comment by nilsdoehring** [comment]
@pq Thanks so much for this!

**Pull request merged**